core-web/libs/sdk/experiments/src/lib/components/DotLayoutWithExperiments.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { ReactNode, Suspense } from 'react';
+
+import { useExperimentVariant } from '../hooks/useExperimentVariant';
+
+export interface PageProviderProps {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    readonly entity: any;
+    readonly children: ReactNode;
+}
+
+/**
+ * Renders a DotCMS layout with experiments.
+ * This component uses 'useExperimentVariant' to determine whether to hide the content because
+ * it has an assigned variant that's different from the currently displayed one.
+ * If so, the component hides the content and an internal effect handles the redirect.
+ * The next time the check occurs, if the assigned variant matches the displayed one,
+ * the component displays the content.
+ *
+ * @param {PageProviderProps} props - The properties for rendering the DotCMS layout.
+ * @returns {JSX.Element} The JSX element representing the DotCMS layout enhanced with experiments.
+ * */
+
+export function DotLayoutWithExperiments(props: PageProviderProps): JSX.Element | null {
+    const { entity, children } = props;
+
+    const { shouldWaitForVariant } = useExperimentVariant(entity);
+
+    return <Suspense>{shouldWaitForVariant() ? null : children}</Suspense>;
```

Comment:
Basically you're making this page client side rendering.

<img width="1485" alt="image" src="https://github.com/dotCMS/core/assets/751424/31ba66a5-5580-4f3d-a200-661d46fc8abd">

Reply:
I made a few changes here. I add the original content with `visibility: hidden`
<img width="633" alt="Screenshot 2024-04-26 at 12 16 58 PM" src="https://github.com/dotCMS/core/assets/1909643/872e956c-fb53-4a0e-843a-494c90ed1cfc">